src/components/withDrawerState.js

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {useIsDrawerOpen} from '@react-navigation/drawer';
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+
+export const withDrawerPropTypes = {
+    isDrawerOpen: PropTypes.bool.isRequired,
+};
+
+export default function withDrawerState(WrappedComponent) {
+    const HOC_Wrapper = (props) => {
+        const isDrawerOpen = useIsDrawerOpen();
```

Comment:
I think I know what you would say about the hook usage, but please check how simple it is compared to this: https://github.com/Expensify/Expensify.cash/pull/2221/commits/8a6df1ba02e2f09fb44517373e6515207eb79b33

Anyway I can swap it for the changes in the diff above

Reply:
At this point, we are using the hook, but wrapping them in an HOC which makes us wonder _why not_ use the hook. Let's see if we can resolve it [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1617649919372600).

Reply:
We still might need this wrap because class components cannot use hooks directly 
Currently this is only used in `ReportView` which became stateless func component and can use `useIsDrawerOpen` but if I want to add it to `ReportActionCompose` for example, I'll need a HOC

Reply:
Cool, I think that's fine. And we have updated the style guide.